libcudacxx/include/cuda/std/__bit/ctz.h

```diff
@@ -30,125 +30,73 @@
 
 _LIBCUDACXX_BEGIN_NAMESPACE_STD
 
-_LIBCUDACXX_HIDE_FROM_ABI constexpr int __binary_ctz2(uint64_t __x, int __c) noexcept
-{
-  return __c + !(__x & 0x1);
-}
-_LIBCUDACXX_HIDE_FROM_ABI constexpr int __binary_ctz4(uint64_t __x, int __c) noexcept
-{
-  return __binary_ctz2(__x >> 2 * !(__x & 0x3), __c + 2 * !(__x & 0x3));
-}
-_LIBCUDACXX_HIDE_FROM_ABI constexpr int __binary_ctz8(uint64_t __x, int __c) noexcept
-{
-  return __binary_ctz4(__x >> 4 * !(__x & 0x0F), __c + 4 * !(__x & 0x0F));
-}
-_LIBCUDACXX_HIDE_FROM_ABI constexpr int __binary_ctz16(uint64_t __x, int __c) noexcept
-{
-  return __binary_ctz8(__x >> 8 * !(__x & 0x00FF), __c + 8 * !(__x & 0x00FF));
-}
-_LIBCUDACXX_HIDE_FROM_ABI constexpr int __binary_ctz32(uint64_t __x, int __c) noexcept
-{
-  return __binary_ctz16(__x >> 16 * !(__x & 0x0000FFFF), __c + 16 * !(__x & 0x0000FFFF));
-}
-_LIBCUDACXX_HIDE_FROM_ABI constexpr int __binary_ctz64(uint64_t __x) noexcept
-{
-  return __binary_ctz32(__x >> 32 * !(__x & 0x00000000FFFFFFFF), 32 * !(__x & 0x00000000FFFFFFFF));
-}
-
-#if !_CCCL_COMPILER(MSVC)
-
 _LIBCUDACXX_HIDE_FROM_ABI constexpr int __constexpr_ctz(uint32_t __x) noexcept
 {
-#  if defined(__CUDA_ARCH__)
-  return __binary_ctz32(static_cast<uint64_t>(__x), 0); // no device constexpr builtins
-#  else
-  return __builtin_ctz(__x);
-#  endif
+  for (int __i = 0; __i < 32; ++__i)
+  {
+    if (__x & (uint32_t{1} << __i))
+    {
+      return __i;
+    }
+  }
+  return 32;
 }
```

Comment:
These functions have been historically split into one line statements due to C++11 constexpr support.

We should just make a clean implementation now that C++11 is dropped and not split it up anymore